pkg/ccl/backupccl/backup_job.go

```diff
@@ -664,18 +681,229 @@ func (b *backupResumer) readManifestOnResume(
 	return &desc, nil
 }
 
+func getSpansProtectedByBackup(
+	ctx context.Context, backupDetails jobspb.BackupDetails, txn *kv.Txn, exec *sql.ExecutorConfig,
+) ([]roachpb.Span, error) {
+	if backupDetails.ProtectedTimestampRecord == nil {
+		return nil, nil
+	}
+
+	ptsRecord, err := exec.ProtectedTimestampProvider.GetRecord(ctx, txn,
+		*backupDetails.ProtectedTimestampRecord)
+	if err != nil {
+		return nil, err
+	}
+
+	return ptsRecord.Spans, nil
+}
+
+func updateTimestampRecordForSchedule(
```

Comment:
nit: A comment re-iterating that this is how we handle PTS chaining for incremental backups would be useful. As another nit, I find it more readable to have the methods listed in order that they're called and the "helper" methods more seperated from the ones doing the brunt of the work. Something like

```
maybeUpdateProtectedTimestampRecordOnSchedule
manageFullBackupPTSChaining
updateTimestampRecordForSchedule

getSpansProtectedByBackup
protectTimestampRecordForSchedule
releaseProtectedTimestampRecordForSchedule
```

This PTS chaining also feels special cased enough that I could see it warranting its own file.

Reply:
very good point, split out into its own file.
Also renamed `updateTimestampRecordForSchedule` to `manageIncrementalBackupPTSChaining`